modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/utils/modules.tsx

```diff
@@ -0,0 +1,135 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import {loadFDSCellRenderer} from '@liferay/client-extension-web';
+import React, {RefObject} from 'react';
+
+import type {FDSCellRendererArgs} from '@liferay/js-api/data-set';
+
+export async function getLiferayJsModule(moduleURL: string): Promise<any> {
+	if (moduleURL.includes(' from ')) {
+		const parts = moduleURL.split(' from ');
+		const moduleName = parts[1].trim();
+
+		let symbolName = parts[0].trim();
+
+		if (
+			symbolName !== 'default' &&
+			(!symbolName.startsWith('{') || !symbolName.endsWith('}'))
+		) {
+			throw new Error(`Invalid module URL: ${moduleURL}`);
+		}
+
+		if (symbolName.startsWith('{')) {
+			symbolName = symbolName.substring(1, symbolName.length - 1).trim();
+		}
+
+		try {
+			const renderer = await loadFDSCellRenderer(moduleName, symbolName);
```

Comment:
The FDS uses this API to load different types of stuff, even now that CE are not integrated with it, namely:
  * [cellRenderers](https://github.com/liferay/liferay-portal/blob/c7bbb6686c4b25fc2c1b983c6cbb23b160c2a96b/modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/utils/dataRenderers.js#L40-L60), the one we're working on in this PR
  * [filters](https://github.com/liferay/liferay-portal/blob/61ab9c110a2e4bac68b246398829222c789363ab/modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/management_bar/components/filters/Filter.js#L75-L81), not yet covered by CE
  * [content renderers](https://github.com/liferay/liferay-portal/blob/f79018e8da5a4c2c8f2bd6008b4429d0aec45f10/modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/FrontendDataSet.js#L362-L378) cover the full data display space, ootb we have cards, list, table and some more. Not yet covered by CE

As a result, the specifics related to loading a FDSCellRenderer CE can just be a particular  usage of this API, which in turn needs support for  specific functions like `loadFDSCellRenderer` provided by the new package.

Perhaps a simpler alternative is to stop using this API (let it be just an AMD-based mechanism) and extend specific CellRenderer loading logic in `dataRenderers.js` to make it able to load a CE CellRenderer. Then, the user of `dataRenderers.js` API ([TableCell](https://github.com/liferay/liferay-portal/blob/61145076b40ca320d45e5cb0a29bf37e7e51bf7e/modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/views/table/TableCell.js#L107-L123) in this case) would wire the right loading mechanism (AMD vs `loadFDSCellRenderer()` function) depending on the config given via `contentRenderer` prop.

This way, we decouple the different CE loading mechanisms from the AMD-based ones and allow each injection point (existing and future ones) to (i) receive a configuration, (ii) invoke the right loading logic depending on it

@markocikos I'd like to know yout thoughts on this

Reply:
> we're artificially introducing an indirection layer

That's on purpose, because we want:

1. Portal developers to use the types from `@liferay/js-api`
2. Have a place where we can implement backward compatibility if we need to

In any case, I'm sure we will evolve these APIs with the time. But we should agree on whether we want to add this seam (a seam is a place where you can alter behavior in your program without editing in that place; see https://www.informit.com/articles/article.aspx?p=359417&seqNum=2) or not.

Reply:
> Perhaps a simpler alternative is to stop using this API (let modules.tsx be just an AMD-based mechanism)

No, because we want to stop using AMD and move everything to ESM. In fact, all this new APIs are not supported as AMD, IINM. At least, if they are it's because our compatibility layer is quite good, because I tried it to be unavailable for AMD on purpose :sweat_smile: .

Reply:
In any case, I'm not sure I understand completely what you are proposing @dsanz , so I'll resort to Slack to clarify it.